src/UAEventEmitter.ts

```diff
@@ -22,9 +27,12 @@ export class UAEventEmitter {
         }
       );
     }
-
-    this.eventEmitter.addListener('com.airship.pending_events', async () => {
-      return this.dispatchEvents(false);
+    this.eventEmitter.addListener('com.airship.ready', () => {
+      return new Promise((resolve, reject) => {
+        InteractionManager.runAfterInteractions(() => {
```

Comment:
Any downsides to this? Did you test that this still works when the app is woken up in the background but not in the foreground, such as a content-available push? If not I can give it a go

Reply:
No downsides. It's basically like calling a timeout but we're just waiting for the JavaScript to be ready. It's probably milliseconds difference.